### ISSUE TITLE
Flannel allows you to send invalid/unused commandline args without dying.

### ISSUE THREAD

**Issue by jayunit100** [coreos/flannel#217]
**UPDATE** when i created this issue, I was not using the proper current `--iface` option for flannel.   Flannel did not die, so i assumed it was reading in the option, but actually, it was just silently ignoring it.... Nevertheless, the reason i discovered this was that I found eth0 wasn't working (its a bad idea to use eth0 for vagrant based VBox setups where, for example, the bridge is eth0).,.... neverthelss ill leave the comments below unaltered for historical purposes.

**but the only bug here is that we need to make sure flannel dies when you give it bad cmd line args** :) ...

Related to #216, I've found that in my flannel subnet, I have a subnet w/ publicIP 10.0.2.15.

```[root@kube0 vagrant]# etcdctl get /coreos.com/network/subnets/99.0.30.0-24
{"PublicIP":"10.0.2.15","BackendType":"vxlan","BackendData":{"VtepMAC":"8a:eb:23:bd:1b:4a"}}```

However, looking deeper into the subnet, the public IP above (10.0.2.15) corresponds to **eth0**.

```[root@kube0 vagrant]# ip a | grep eth1
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    inet 192.168.4.100/24 brd 192.168.4.255 scope global eth1```

Oddly, though, I have launched flannel with the eth1 argument: 

```
[root@kube0 vagrant]# ps -ax | grep fla
16332 ?        Ssl    0:00 /usr/bin/flanneld -etcd-endpoints=http://127.0.0.1:2379 -etcd-prefix=/coreos.com/network eth1
```

I'm assuming that normally, the flannel publicIP should always match the **eth** interface which flannel is launched on, but maybe there are situations where flannel **falls back** to eth0 even if another nic is specified?

**Comment by jayunit100** [comment]
Diving deeper, it appears that the IP Addresses in the flannel subnet aren't actually written to etcd anywhere, instead, I only see 

```
[root@kube0 vagrant]# etcdctl ls /coreos.com/network/subnets/
/coreos.com/network/subnets/99.0.30.0-24
```

However, I've already allocated a few containers.  

- Should each container have an entry somewhere in ETCD for its IP address? 
- Is flannel responsible, on startup, to ensure that IP allocations will indeed be recorded properly in ETCD?

If so, maybe this is a "bug" of some sort, i.e. the fact that its possible for docker to allocate flannel  addresses **without** registering them in etcd?

**Comment by jayunit100** [comment]
**NEVERMIND** I foud that the **eth1** argument being sent requires a `--iface=` option. Meanwhile, i still have other issues, but now im getting two subnets properly provisioned by flannel to different VM's container spaces.

For anyone curious, you have to be careful to make sure you properly use the iface option when starting flannel with vagrant provisioned VMs on a bridged network (eth0 is the bridge in some cases, which can lead to two separate machines trying to use IPs in the same subnet)

**Comment by jayunit100** [comment]
REopening because, flannel should provide an error if you give it an unused/invalid commandline option.     Will change the name shortly...

**Comment by eyakubovich** [comment]
I think https://github.com/coreos/flannel/pull/223 fixes this. Can it be closed? With latest Go (1.4), bad command line options should result in an error.
